erc/ERCS/erc-7562.md

```diff
@@ -179,44 +274,49 @@ To help make sense of these params, note that a malicious paymaster can at most
-    are treated exactly like persistent storage (SLOAD/SSTORE)
+        * Only allow known accepted precompiles on the network, that do not access anything in the blockchain state or environment.  
+        * The core precompiles 0x1 .. 0x9
+        * The [RIP-7212](./rip-7212.md) sec256r1 precompile, on networks that accepted it.
+* **[OP-070]** Transient Storage slots defined in [EIP-1153](./eip-1153) and accessed using `TLOAD` (`0x5c`) and `TSTORE` (`0x5d`) opcodes
+  are treated exactly like persistent storage (SLOAD/SSTORE).
 
 ### Code Rules
 
 * **[COD-010]** Between the first and the second validations, the `EXTCODEHASH` value of any visited address,
-entity or referenced library, may not be changed.\
-If the code is modified, the UserOperation is considered invalid.
+  entity, or referenced library, may not be changed.\
+  If the code is modified, the UserOperation is considered invalid.
 
 ### Storage rules.
 
-The permanent storage access with `SLOAD` and `SSTORE` instructions within each phase are limited as follows:
+The permanent storage access with `SLOAD` and `SSTORE` instructions within each phase is limited as follows:
 
 * **[STO-010]** Access to the "account" storage is always allowed.
-* Access to associated storage of the account in an external (non-entity contract) is allowed if either:
-    * **[STO-021]**  The account already exists
+* Access to associated storage of the account in an external (non-entity) contract is allowed if either:
+    * **[STO-021]**  The account already exists.
     * **[STO-022]**  There is an `initCode` and the `factory` contract is staked.
 * If the entity (`paymaster`, `factory`) is staked, then it is also allowed:
     * **[STO-031]** Access the entity's own storage.
-    * **[STO-032]** Read/Write Access to storage slots that is associated with the entity, in any non-entity contract.
+    * **[STO-032]** Read/Write Access to storage slots that are associated with the entity, in any non-entity contract.
     * **[STO-033]** Read-only access to any storage in non-entity contract.
+
+### Local Rules
+
+These are storage rules that help to protect the bundler from denial of service:
 * **[STO-040]** `UserOperation` may not use an entity address (`factory`/`paymaster`/`aggregator`) that is used as an "account" in another `UserOperation` in the mempool. \
-This means that a `Paymaster` and `Factory` contracts cannot practically be an "account" contract as well.
-* **[STO-041]** A contract whose Associated Storage slot is accessed during a `UserOperation` may not be an address of a "sender" in another `UserOperation` in the mempool.
+  This means that `Paymaster` and `Factory` contracts cannot practically be an "account" contract as well.
+* **[STO-041]** A contract whose storage is associated with any entity of a `UserOperation` may not be a "sender" of another UserOperation in the mempool.
```

Comment:
Somehow my comment about this keeps disappearing as things move around.  I think this rule is **still** broken.

The way I'd interpret the rule as written above:
"If the mempool has op1 and op2, and op1 accesses someContract.balances[op2.sender], exclude **op2**".
which allows op1 to censor op2.

The intended behavior is the opposite:
"If the mempool has op1 and op2, and op1 accesses someContract.balances[op2.sender], exclude **op1**".

There is no concern that op2 will censor op1 because it requires the attacker to control the address and have it pass validation. But the opposite is not true - op1 can easily censor op2 if it is given priority.

Reply:
rephrased as
`UserOperation` may not use associated storage (of either its account or from staked entity) in a contract that is a "sender" of another UserOperation in the mempool.